### PR TITLE
Allow empty/missing SLM retention configuration

### PR DESCRIPTION
**Pull request by dakrone**
Semi-related to #44465, this allows the `"retention"` configuration map
to be missing.

Relates to #43663

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java [hunk 1]
```diff
@@ -102,7 +102,7 @@ public SnapshotLifecyclePolicy(StreamInput in) throws IOException {
         this.schedule = in.readString();
         this.repository = in.readString();
         this.configuration = in.readMap();
-        this.retentionPolicy = new SnapshotRetentionConfiguration(in);
+        this.retentionPolicy = in.readOptionalWriteable(SnapshotRetentionConfiguration::new);
```
Comment (gwbrown) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java]: I didn't notice this the first time around but we'll probably need to version-gate this if there's a version of SLM that ships that doesn't have retention.

**Review by AthenaEryma — APPROVED** [review]
LGTM, left one comment that should be addressed but not necessarily in this PR.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java (hunk 1 above)
Comment (dakrone) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/slm/SnapshotLifecyclePolicy.java]: +1 on addressing this, but probably not until we're about to merge

**Comment by dakrone** [comment]
@elasticmachine update branch

**Pull request merged**